products-service/handlers/get-product-by-id.ts

```diff
@@ -0,0 +1,36 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import * as products from '../products/products.json';
+import 'source-map-support/register';
+import { corsHeaders, errorMessages } from '../utils';
+
+export const getProductById: APIGatewayProxyHandler = async (event, _context) => {
+    try {
+        const { productId } = event.pathParameters;
+        if (productId) {
+            const product = products.find(({ id }) => productId === id);
+            if (product) {
+                return {
+                    statusCode: 200,
+                    headers: corsHeaders,
+                    body: JSON.stringify(product, null, 2),
+                };
+            }
+            return {
+                statusCode: 404,
+                headers: corsHeaders,
+                body: errorMessages.notFound(productId),
+            }
+        }
+        return {
+            statusCode: 400,
```

Comment:
This error will never be triggered. If you send a request to your `products/{id}` route without an `{id}` param, you'll simply trigger a `/products` route handler, which will return you a list of products.

Reply:
Agree, but no one can guarantee that endpoint for getProductById will not change, but I can guarantee that after that change we will forget to add error handling.